test/integration/plugin_test.go

```diff
@@ -0,0 +1,528 @@
+package integration
+
+import (
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/kubernetes-incubator/kube-aws/core/root"
+	"github.com/kubernetes-incubator/kube-aws/core/root/config"
+	"github.com/kubernetes-incubator/kube-aws/model"
+	"github.com/kubernetes-incubator/kube-aws/plugin"
+	"github.com/kubernetes-incubator/kube-aws/test/helper"
+)
+
+func TestPlugin(t *testing.T) {
+	kubeAwsSettings := newKubeAwsSettingsFromEnv(t)
+
+	s3URI, s3URIExists := os.LookupEnv("KUBE_AWS_S3_DIR_URI")
+
+	if !s3URIExists || s3URI == "" {
+		s3URI = "s3://examplebucket/exampledir"
+		t.Logf(`Falling back s3URI to a stub value "%s" for tests of validating stack templates. No assets will actually be uploaded to S3`, s3URI)
+	}
+
+	mainClusterYaml := kubeAwsSettings.mainClusterYaml()
+	minimalValidConfigYaml := mainClusterYaml + `
+availabilityZone: us-west-1c
+`
+	validCases := []struct {
+		context       string
+		clusterYaml   string
+		plugins       []helper.TestPlugin
+		assertConfig  []ConfigTester
+		assertCluster []ClusterTester
+	}{
+		{
+			context: "WithAddons",
+			clusterYaml: minimalValidConfigYaml + `
+kubeAwsPlugins:
+  myPlugin:
+    enabled: true
+    queue:
+      name: baz1
+    oidc:
+      issuer:
+        url: "https://login.example.com/"
+
+worker:
+  nodePools:
+  - name: pool1
+    kubeAwsPlugins:
```

Comment:
I'd like to rename this to just `plugins` if we could agree with renaming the current `plugins` used solely for `rbac` to something else(perhaps `kubernetes` and `kubernetes.rbac` so that it nicely represents the fact it is used to enable `rbac` across the k8s cluster?)